components/builder-api/src/server/resources/origins.rs

```diff
@@ -867,32 +755,19 @@ fn upload_origin_secret_key(req: HttpRequest,
             debug!("Can't parse secret key upload content: {}", e);
             return HttpResponse::new(StatusCode::UNPROCESSABLE_ENTITY);
         }
-    }
-
-    let (encrypted_body, bldr_key_rev) = match encrypt(&req, &body) {
-        Ok((encrypted, rev)) => (encrypted, rev),
-        Err(err) => {
-            debug!("Failed to encrypt body, err={:?}", err);
-            return err.into();
-        }
     };
 
-    let new_sk = NewOriginPrivateSigningKey { owner_id:           account_id as i64,
-                                              origin:             &origin,
-                                              name:               &origin,
-                                              full_name:          &format!("{}-{}",
-                                                                           &origin, &revision),
-                                              revision:           &revision,
-                                              body:               &encrypted_body.as_bytes(),
-                                              encryption_key_rev: &bldr_key_rev, };
-
-    match OriginPrivateSigningKey::create(&new_sk, &*conn).map_err(Error::DieselError) {
-        Ok(_) => HttpResponse::Created().finish(),
-        Err(err) => {
-            debug!("{}", err);
-            err.into()
-        }
+    if let Err(e) = save_secret_origin_signing_key(account_id,
+                                                   &origin,
+                                                   &state.config.api.key_path,
+                                                   &key,
+                                                   &*conn)
+    {
+        error!("create_keys: Failed to create private key, err={}", e);
```

Comment:
Trivial but is it worth clarifying in the error message here what type of private key we failed to create? Might also be good to include the origin the key was associated with?

Reply:
Can do!